### PR TITLE
Update ruff and fix type error

### PR DESCRIPTION
**Pull request by alexrudd2**

### CODE REVIEW

File: pymodbus_repl/lib/helper.py [hunk 1]
```diff
@@ -74,7 +74,7 @@
 )
 
 
-class Command:
+class Command(dict):
```
Comment (dhoomakethu) [inline pymodbus_repl/lib/helper.py:77]: Why do we need to derive from `dict` ?

File: pymodbus_repl/server/cli.py [hunk 2]
```diff
@@ -17,6 +17,9 @@
 from pymodbus_repl import __VERSION__ as repl_version
 from pymodbus_repl.lib.completer import CmdCompleter
 from pymodbus_repl.lib.helper import Command, style
+from pymodbus_repl.lib.reactive import (
+    ReactiveServer,
```
Comment (dhoomakethu) [inline pymodbus_repl/server/cli.py:21]: If I remember this correctly, this was causing a circular import error. I am good if this is not the case any more.

**Comment by dhoomakethu** [comment]
Thanks for the PR @alexrudd2

File: pymodbus_repl/lib/helper.py (hunk 1 above)
Comment (alexrudd2) [reply pymodbus_repl/lib/helper.py:77]: It makes mypy shut up about this line:

https://github.com/pymodbus-dev/repl/blob/0808283e9adc323c3d54d345468cda73b399d2ec/pymodbus_repl/server/cli.py#L120

The other options are:
* `type: ignore`
* make `COMMANDS` actually `dict[str, Command]`

Comment (dhoomakethu) [reply pymodbus_repl/lib/helper.py:77]: `COMMANDS` is an actual `dict` refer https://github.com/pymodbus-dev/repl/blob/0808283e9adc323c3d54d345468cda73b399d2ec/pymodbus_repl/server/cli.py#L40

File: pymodbus_repl/server/cli.py (hunk 2 above)
Comment (alexrudd2) [reply pymodbus_repl/server/cli.py:21]: I don't see any such error (but didn't look very closely).  Does it work in your tests?

Comment (dhoomakethu) [reply pymodbus_repl/server/cli.py:21]: Looks good.

**Review by dhoomakethu — APPROVED** [review]

**Comment by alexrudd2** [comment]
@dhoomakethu Did you intend to merge this?  (I don't have the commit bit)

**Comment by dhoomakethu** [comment]
Sorry @alexrudd2 for the delay. Merging it now

**Pull request merged**